### PR TITLE
2.0.0 master PR

### PR DESCRIPTION
**Pull request by zeibura**
Master PR for Code4z v2.0.0
Not ready yet - opening to check the diffs.

### CODE REVIEW

**Comment by zeibura** [comment]
Ready for review now

File: README.md [hunk 1]
```diff
@@ -2,34 +2,66 @@
 
 [![License](https://img.shields.io/badge/license-BROADCOM-brightgreen)](https://github.com/BroadcomMFD/code4z/blob/master/LICENSE)
 
-Code4z is an all-in-one VS Code extension pack for mainframe users working with z/OS. Code4z provides language support for IBM COBOL 6.0 and High Level Assembler for z/OS. It includes Zowe Explorer, Explorer for Endevor and extensions for interactive debugging and graphical visualization of COBOL applications. Code4z is free to install and use.
+Code4z is an all-in-one VS Code extension pack for mainframe users working with z/OS. Code4z provides language support for COBOL and High Level Assembler for z/OS and graphical visualization of COBOL applications. It also includes data editing and file management tools, Explorer for Endevor, and extensions for interactive debugging, macro tracing and abend analysis. Code4z is free to install and use.
```
Comment (IgorCATech) [inline README.md:5]:
```suggestion
Code4z is an all-in-one VS Code extension pack for mainframe users working with z/OS. Code4z provides language support for COBOL and High Level Assembler for z/OS and graphical visualization of COBOL applications. The pack also includes data editing and file management tools, Explorer for Endevor, and extensions for interactive debugging, macro tracing, and abend analysis. Code4z is free to install and use.

```

File: README.md [hunk 2]
```diff
@@ -2,34 +2,66 @@
...
+Code4z is compatible with Microsoft Visual Studio Code and Github Codespaces.
+
+<div align="center">
+  <img src="https://raw.githubusercontent.com/BroadcomMFD/code4z/master/extensions4.png" alt="Code4z Extensions" />
+</div>
 
 ## How Does It Work?
 
-Code4z extensions utilize language servers that are fully compatible with the [Language Server Protocol Specification](https://microsoft.github.io/language-server-protocol/specification). Additional editor extensions simplify your common workflows and enable you to work with COBOL and HLASM code in the same way you work with other languages in Visual Studio Code.
+The Code4z extension pack simplifies your common workflows and enables you to work with COBOL and HLASM code in the same way you work with other languages in Visual Studio Code. The pack provides a modern mainframe programming experience and includes tools for:
```
Comment (IgorCATech) [inline README.md:15]:
```suggestion
The Code4z extension pack simplifies your common workflows and enables you to work with COBOL and HLASM code in the same way you work with other languages in Visual Studio Code. The pack provides a modern mainframe programming experience and includes the following tools:

```

File: README.md [hunk 3]
```diff
@@ -2,34 +2,66 @@
...
-- Utilize graphical visualization of program flow for COBOL programs.
-- Automatically retrieve copybooks from the mainframe.
-- Trace HLASM macros.
-- Retrieve, browse and search Endevor elements.
-- Access z/OS Datasets and z/OS Unix file systems, and submit JCLs.
-- Debug COBOL code for applications running in a CICS region.
-- View and download job output.
-- Issue TSO and MVS commands.
+- **Language support** for IBM Enterprise COBOL 6.0 and high-level assembler language. 
+  - The COBOL Language Support and HLASM Language Support extensions provide syntax highlighting, syntax validation and content assist when editing COBOL and HLASM code. Code4z extensions utilize language servers that are fully compatible with the [Language Server Protocol Specification](https://microsoft.github.io/language-server-protocol/specification).
```
Comment (IgorCATech) [inline README.md:18]:
```suggestion
  - The COBOL Language Support and HLASM Language Support extensions provide syntax highlighting, syntax validation, and content assistance when editing COBOL and HLASM code. Code4z extensions utilize language servers that are fully compatible with the [Language Server Protocol Specification](https://microsoft.github.io/language-server-protocol/specification). 

```

File: README.md [hunk 4]
```diff
@@ -2,34 +2,66 @@
...
-- Access z/OS Datasets and z/OS Unix file systems, and submit JCLs.
-- Debug COBOL code for applications running in a CICS region.
-- View and download job output.
-- Issue TSO and MVS commands.
+- **Language support** for IBM Enterprise COBOL 6.0 and high-level assembler language. 
+  - The COBOL Language Support and HLASM Language Support extensions provide syntax highlighting, syntax validation and content assist when editing COBOL and HLASM code. Code4z extensions utilize language servers that are fully compatible with the [Language Server Protocol Specification](https://microsoft.github.io/language-server-protocol/specification). 
+  - You can configure COBOL Language Support to automatically download your copybooks from the mainframe. 
+  - The COBOL Control Flow application provides graphical visualization of program flow for COBOL programs.
+- **Data editing and file management** of mainframe data sets. 
+  - With Zowe Explorer you can allocate, copy, rename and edit mainframe data sets in VS Code and save them to the mainframe.
```
Comment (IgorCATech) [inline README.md:22]:
```suggestion
  - With Zowe Explorer you can allocate, copy, rename, and edit mainframe data sets in VS Code and save them to the mainframe. 

```

File: README.md [hunk 5]
```diff
@@ -2,34 +2,66 @@
...
-- Debug COBOL code for applications running in a CICS region.
-- View and download job output.
-- Issue TSO and MVS commands.
+- **Language support** for IBM Enterprise COBOL 6.0 and high-level assembler language. 
+  - The COBOL Language Support and HLASM Language Support extensions provide syntax highlighting, syntax validation and content assist when editing COBOL and HLASM code. Code4z extensions utilize language servers that are fully compatible with the [Language Server Protocol Specification](https://microsoft.github.io/language-server-protocol/specification). 
+  - You can configure COBOL Language Support to automatically download your copybooks from the mainframe. 
+  - The COBOL Control Flow application provides graphical visualization of program flow for COBOL programs.
+- **Data editing and file management** of mainframe data sets. 
+  - With Zowe Explorer you can allocate, copy, rename and edit mainframe data sets in VS Code and save them to the mainframe. 
+  - The Data Editor for Mainframe extension allows you to edit VSAM data sets, apply record layouts, and filter records using selection criteria.
```
Comment (IgorCATech) [inline README.md:23]:
```suggestion
  - The Data Editor for Mainframe extension enables you to edit VSAM data sets, apply record layouts, and filter records using selection criteria.

```

File: README.md [hunk 6]
```diff
@@ -2,34 +2,66 @@
...
+- **Testing tools** for CICS and Batch programs. 
+  - The Debugger for Mainframe extension lets you debug CICS and Batch applications written in COBOL.
```
Comment (IgorCATech) [inline README.md:25]:
```suggestion
  - The Debugger for Mainframe extension enables you to debug CICS and Batch applications that are written in COBOL. 

```

File: README.md [hunk 7]
```diff
@@ -2,34 +2,66 @@
...
+  - You can configure COBOL Language Support to automatically download your copybooks from the mainframe. 
+  - The COBOL Control Flow application provides graphical visualization of program flow for COBOL programs.
+- **Data editing and file management** of mainframe data sets. 
+  - With Zowe Explorer you can allocate, copy, rename and edit mainframe data sets in VS Code and save them to the mainframe. 
+  - The Data Editor for Mainframe extension allows you to edit VSAM data sets, apply record layouts, and filter records using selection criteria.
+- **Testing tools** for CICS and Batch programs. 
+  - The Debugger for Mainframe extension lets you debug CICS and Batch applications written in COBOL. 
+  - With Abend Analyzer for Mainframe you can view abend reports and symbolic data. 
+  - The HLASM Language Support extension lets you trace Assembler macros interactively.
```
Comment (IgorCATech) [inline README.md:27]:
```suggestion
  - The HLASM Language Support extension enables you to trace Assembler macros interactively.

```

File: README.md [hunk 8]
```diff
@@ -2,34 +2,66 @@
...
+- **Source code management** integrated with Endevor. 
+  - Using the Explorer for Endevor extension you can retrieve, browse and search Endevor elements in your IDE.
```
Comment (IgorCATech) [inline README.md:29]:
```suggestion
  - With the help of the Explorer for Endevor extension you can retrieve, browse, and search Endevor elements in your IDE.

```

File: README.md [hunk 9]
```diff
@@ -2,34 +2,66 @@
...
+- **Mainframe operations**.
+  - With Zowe Explorer you can issue TSO and MVS commands, and submit JCLs.
```
Comment (IgorCATech) [inline README.md:31]:
```suggestion
  - With Zowe Explorer you can issue TSO and MVS commands and submit JCLs.

```

File: README.md [hunk 10]
```diff
@@ -2,34 +2,66 @@
...
+
+### Sample Workflows
+
+#### Edit COBOL and HLASM source code stored on the mainframe
+
+* Use VS Code’s built-in Git interface to clone source code from Endevor using Endevor’s Bridge for Git.
```
Comment (IgorCATech) [inline README.md:37]:
```suggestion
* Use the VS Code built-in Git interface to clone source code from Endevor, using Endevor Bridge for Git.

```

File: README.md [hunk 11]
```diff
@@ -2,34 +2,66 @@
...
+* Edit COBOL code using COBOL Language Support, taking advantage of all the coding assistance provided by the extension.
```
Comment (IgorCATech) [inline README.md:38]:
```suggestion
* Edit COBOL code using COBOL Language Support to enjoy all the coding assistance that is provided by the extension.

```

File: README.md [hunk 12]
```diff
@@ -2,34 +2,66 @@
...
+* Use COBOL Control Flow to easily navigate around different parts of the source code.
+* Edit assembler code and trace macros using HLASM Language Support.
+* Locate all copybooks in the Git folder structure, and if needed, download other copybooks used in the source code from the mainframe.
```
Comment (IgorCATech) [inline README.md:41]:
```suggestion
* Locate all copybooks in the Git folder structure and, if needed, download other copybooks that are used in the source code from the mainframe.

```

File: README.md [hunk 13]
```diff
@@ -2,34 +2,66 @@
...
+* Use VS Code’s built-in Git interface to push, pull and merge changes.
```
Comment (IgorCATech) [inline README.md:42]:
```suggestion
* Use the VS Code built-in Git interface to push, pull, and merge changes.

```

File: README.md [hunk 14]
```diff
@@ -2,34 +2,66 @@
...
+* Search and browse additional code elements in the entire map of Endevor using the Explorer for Endevor extension. The COBOL Language Support extension provides syntax awareness for COBOL elements.
```
Comment (IgorCATech) [inline README.md:43]:
```suggestion
* Search and browse additional code elements in the entire map of Endevor, using the Explorer for Endevor extension. The COBOL Language Support extension provides syntax awareness for COBOL elements.

```

File: README.md [hunk 15]
```diff
@@ -2,34 +2,66 @@
...
+* Download code elements, including their dependencies, into the current workspace using the Explorer for Endevor extension, and synchronize them with the Git project if code modifications are needed.
```
Comment (IgorCATech) [inline README.md:44]:
```suggestion
* Download code elements, including their dependencies, into the current workspace, using Explorer for Endevor, and synchronize them with the Git project if code modifications are needed.

```

File: README.md [hunk 16]
```diff
@@ -41,7 +73,7 @@ With the Code4z extension pack you can:
 [![GitHub issues](https://img.shields.io/github/issues-raw/eclipse/che-che4z-lsp-for-cobol?style=flat-square)](https://github.com/eclipse/che-che4z-lsp-for-cobol/issues)
 [![slack](https://img.shields.io/badge/chat-on%20Slack-blue?style=flat-square)](https://communityinviter.com/apps/che4z/code4z)
 
-COBOL Language Support provides autocomplete, highlighting and diagnostic features for COBOL code and copybooks.
+COBOL Language Support enhances the COBOL programming experience on your IDE. The extension leverages the language server protocol to provide autocomplete, syntax highlighting and coloring, and diagnostic features for COBOL code and copybooks. The COBOL Language Support extension can also connect to a mainframe using the Zowe Explorer extension to automatically retrieve copybooks used in your programs and store them in your workspace.
```
Comment (IgorCATech) [inline README.md:88]:
```suggestion
COBOL Language Support enhances the COBOL programming experience on your IDE. The extension leverages the language server protocol to provide autocomplete, syntax highlighting and coloring, and diagnostic features for COBOL code and copybooks. The COBOL Language Support extension can also connect to a mainframe using the Zowe Explorer extension to automatically retrieve copybooks that are used in your programs and store them in your workspace.

```

File: README.md [hunk 17]
```diff
@@ -53,14 +85,30 @@ COBOL Language Support provides autocomplete, highlighting and diagnostic featur
 * [Beginner’s Guide: COBOL Made Easy](https://medium.com/modern-mainframe/beginners-guide-cobol-made-easy-introduction-ecf2f611ac76)
 * [LSP Magic - Mainframe Language Support in Modern IDEs](https://medium.com/modern-mainframe/lsp-magic-mainframe-language-support-in-modern-ides-4ea3d81259b3)
 
+## [COBOL Control Flow](https://marketplace.visualstudio.com/items?itemName=broadcomMFD.ccf)
+[![Visual Studio Marketplace Installs](https://img.shields.io/visual-studio-marketplace/i/broadcomMFD.ccf?color=brighgreen&label=Marketplace&style=flat-square)](https://marketplace.visualstudio.com/items?itemName=broadcomMFD.ccf)
+[![GitHub issues](https://img.shields.io/github/issues-raw/broadcomMFD/cobol-control-flow?style=flat-square)](https://github.com/BroadcomMFD/cobol-control-flow/issues)
+[![slack](https://img.shields.io/badge/chat-on%20Slack-blue?style=flat-square)](https://communityinviter.com/apps/che4z/code4z)
+
+COBOL Control Flow provides graphical visualization of program flow for programs written in COBOL. The extension is designed to help COBOL developers to quickly comprehend and debug COBOL programs with which they might not be familiar.
```
Comment (IgorCATech) [inline README.md:94]:
```suggestion
COBOL Control Flow provides graphical visualization of program flow for programs that are written in COBOL. The extension is designed to help COBOL developers to quickly comprehend and debug COBOL programs that they might not be familiar with.

```

File: README.md [hunk 18]
```diff
@@ -53,14 +85,30 @@ COBOL Language Support provides autocomplete, highlighting and diagnostic featur
...
 
 ## [HLASM Language Support](https://marketplace.visualstudio.com/items?itemName=broadcomMFD.hlasm-language-support) 
 
 [![Visual Studio Marketplace Installs](https://img.shields.io/visual-studio-marketplace/i/broadcomMFD.hlasm-language-support?color=brighgreen&label=Marketplace&style=flat-square)](https://marketplace.visualstudio.com/items?itemName=broadcomMFD.hlasm-language-support)
 [![GitHub issues](https://img.shields.io/github/issues-raw/eclipse/che-che4z-lsp-for-hlasm?style=flat-square)](https://github.com/eclipse/che-che4z-lsp-for-hlasm/issues)
 [![slack](https://img.shields.io/badge/chat-on%20Slack-blue?style=flat-square)](https://communityinviter.com/apps/che4z/code4z)
 
-Code completion, highlighting, browsing and validation for High Level Assembler language.
+HLASM Language Support is an extension that supports the High Level Assembler language. It provides code completion, highlighting and navigation features, shows mistakes in the source, and lets you trace how the conditional assembly is evaluated with a modern debugging experience.
```
Comment (IgorCATech) [inline README.md:112]:
```suggestion
HLASM Language Support is an extension that supports the High Level Assembler language. The extension provides code completion, highlighting and navigation features, shows mistakes in the source, and enables you to check how the conditional assembly is evaluated with a modern debugging experience.

```

File: README.md [hunk 19]
```diff
@@ -117,6 +165,22 @@ This extension offers best developer experience in synergy with [Bridge for Git]
 - Work with Endevor profiles and Zowe CLI base profiles
 - Print listings
 
+## [Data Editor for Mainframe](https://marketplace.visualstudio.com/items?itemName=broadcomMFD.data-editor-for-mainframe)
+
+The Data Editor for Mainframe extension for VS Code adds a modern user interface to [File Master Plus for MVS](https://www.broadcom.com/products/mainframe/testing-and-quality/file-master-plus). Data Editor for Mainframe allows you to view and edit mainframe data sets, including VSAM data sets, in your IDE.
```
Comment (IgorCATech) [inline README.md:171]:
```suggestion
The Data Editor for Mainframe extension for VS Code adds a modern user interface to [File Master Plus for MVS](https://www.broadcom.com/products/mainframe/testing-and-quality/file-master-plus). Data Editor for Mainframe enables you to view and edit mainframe data sets, including VSAM data sets, in your IDE.

```

File: README.md [hunk 20]
```diff
@@ -117,6 +165,22 @@ This extension offers best developer experience in synergy with [Bridge for Git]
...
 
+## [Data Editor for Mainframe](https://marketplace.visualstudio.com/items?itemName=broadcomMFD.data-editor-for-mainframe)
+
+The Data Editor for Mainframe extension for VS Code adds a modern user interface to [File Master Plus for MVS](https://www.broadcom.com/products/mainframe/testing-and-quality/file-master-plus). Data Editor for Mainframe allows you to view and edit mainframe data sets, including VSAM data sets, in your IDE.
+
+> How can we improve Data Editor for Mainframe? [Let us know on our Git repository](https://github.com/BroadcomMFD/data-editor-for-mainframe/issues)
+
+### Features
+
+- Browse and edit VSAM data sets, sequential data sets and PDS members.
```
Comment (IgorCATech) [inline README.md:177]:
```suggestion
- Browse and edit VSAM data sets, sequential data sets, and PDS members.

```

File: README.md [hunk 21]
```diff
@@ -135,42 +199,17 @@ Debugger for Mainframe provides the debugging interface to [InterTest™ for CIC
...
-[![GitHub issues](https://img.shields.io/github/issues-raw/broadcomMFD/cobol-control-flow?style=flat-square)](https://github.com/BroadcomMFD/cobol-control-flow/issues)
-[![slack](https://img.shields.io/badge/chat-on%20Slack-blue?style=flat-square)](https://communityinviter.com/apps/che4z/code4z)
+## [Abend Analyzer for Mainframe](https://marketplace.visualstudio.com/items?itemName=broadcomMFD.abend-analyzer)
 
-COBOL Control Flow provides graphical visualization of program flow for programs written in COBOL. The extension is designed to help COBOL developers to quickly comprehend and debug COBOL programs with which they might not be familiar.
+Abend Analyzer for Mainframe provides an interface to [SymDump® CICS](https://www.broadcom.com/products/mainframe/testing-and-quality/symdump-cics) and [SymDump® Batch](https://www.broadcom.com/products/mainframe/testing-and-quality/symdump-batch). This extension allows you to browse, manage and view formatted abend reports and symbolic data in a modern IDE environment.
```
Comment (IgorCATech) [inline README.md:205]:
```suggestion
Abend Analyzer for Mainframe provides an interface to [SymDump® CICS](https://www.broadcom.com/products/mainframe/testing-and-quality/symdump-cics) and [SymDump® Batch](https://www.broadcom.com/products/mainframe/testing-and-quality/symdump-batch). The extension enables you to browse, manage, and view formatted abend reports and symbolic data in a modern IDE environment.

```

**Review by IgorCATech — CHANGES_REQUESTED** [review]
I've left a few suggestions

File: README.md [hunk 22]
```diff
@@ -2,34 +2,66 @@
...
+- **Source code management** integrated with Endevor. 
+  - With the help of the Explorer for Endevor extension you can retrieve, browse, and search Endevor elements in your IDE.
+- **Mainframe operations**.
+  - With Zowe Explorer you can issue TSO and MVS commands and submit JCLs.
+
+### Sample Workflows
+
+#### Edit COBOL and HLASM source code stored on the mainframe
+
+* Use the VS Code built-in Git interface to clone source code from Endevor, using Endevor Bridge for Git.
```
Comment (IgorCATech) [inline README.md:37]: I don't like the double `use` here... but I don't have a better suggestion :)

**Review by IgorCATech — APPROVED** [review]
Thank you for addressing the comments. LGTM!

**Pull request merged**